easybuild/tools/modules.py

```diff
@@ -1430,9 +1430,22 @@ def get_software_libdir(name, only_one=True, fs=None):
     res = []
     if root:
         for lib_subdir in lib_subdirs:
-            if os.path.exists(os.path.join(root, lib_subdir)):
-                if fs is None or any([os.path.exists(os.path.join(root, lib_subdir, f)) for f in fs]):
+            lib_dir_path = os.path.join(root, lib_subdir)
+            if os.path.exists(lib_dir_path):
+                # take into account that lib64 could be a symlink to lib (or vice versa)
+                # see https://github.com/easybuilders/easybuild-framework/issues/3139
+                new_lib_dir = True
```

Comment:
IMO the verbosity/log is not required here. Then it can be simplified to `if any(os.path.samefile(lib_dir_path, os.path.join(root, subdir) for subdir in res): continue`

Reply:
The simplified version you propose is not good enough, because the `os.path.exists(lib_dir_path)` check if really needed (calling `os.path.samefile` with a non-existing path is trouble).

I'll change the log statements to `log.debug` though to make it less verbose.

Reply:
Ah, wait, what you propose goes under the `if os.path.exists(lib_dir_path):`, so it does make sense, thanks.